mantidproject/mantid
# Handle versions properly conda build

**Description of work.**
Presently version numbers inside of mantid are not handled very well whilst building using conda-build. Hence adding this support.



**To test:**
You will need mambaforge installed as per our development guides.

Clone conda-recipes (https://github.com/mantidproject/conda-recipes)
Change the recipes/mantid/meta.yaml file's source section to include:
```
git_url: https://github.com/mantidproject/mantid.git
git_tag: handle-versions-properly-conda-build
```
instead of a URL, and sha256 make sure to comment out these lines.
![image](https://user-images.githubusercontent.com/24723609/151327319-39863d1a-d8d7-4b56-8643-3cb6c51a5938.png)
Then conda-build by navigating to conda-recipes/recipes in a terminal
Create a new conda environment using `mamba create -n conda-build boa conda-verify -y`
Activate this environment `mamba activate conda-build`
Then run `conda mambabuild ./mantid` from within the `recipes` directory of the conda-recipes repository
If it finishes successfully then it passes.



*There is no associated issue.*

*This does not require release notes* because **Developer Facing**




---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
